resources/lib/tubecast/youtube/app.py

```diff
@@ -61,13 +65,25 @@ def _initial_app_state(self):
         self.ofs = 0
         self.has_client = False
         self.volume_monitor = None
-        self.listener = None
+        self.__set_listener(None)
         self.connected_client = None
         # Hold references to the index of received codes
         self.code = -1
         # Get service announcement data
         self.bind_vals = templates.announcement(self.screen_uid, self.default_screen_name, self.default_screen_app)
 
+    def __set_listener(self, listener):  # type: (Optional[YoutubeListener]) -> None
```

Comment:
This makes sure that if you disconnect and connect multiple times the previous listeners are stopped.

This might warrant use of a property.

Reply:
This does a bit more than setting the member var. Name might be improved to be more clear. Other option is to decouple this into 2 methods (set and start) keeping the checks

Reply:
How about `__replace_listener` for the name?

I think removing the call to start has little to no benefit though because the listener should always be running anyway.